components/gantry/oneaxis/oneaxis.go

```diff
@@ -37,45 +37,29 @@ type Config struct {
 
 // Validate ensures all parts of the config are valid.
 func (cfg *Config) Validate(path string) ([]string, error) {
-	deps, err := cfg.validate()
-	if err != nil {
-		err = utils.NewConfigValidationError(path, err)
-	}
-	return deps, err
-}
-
-func (cfg *Config) validate() ([]string, error) {
 	var deps []string
 
 	if len(cfg.Motor) == 0 {
-		return nil, errors.New("cannot find motor for gantry")
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "motor")
 	}
 	deps = append(deps, cfg.Motor)
 
 	if cfg.LengthMm <= 0 {
-		return nil, errors.New("each axis needs a non-zero and positive length")
-	}
-
-	if len(cfg.LimitSwitchPins) == 0 && len(cfg.Board) > 0 {
-		return nil, errors.New("gantry with encoders have to assign boards or controllers to motors")
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "length_mm")
```

Comment:
I'm open to disagreement, but I liked the old error more: the new one makes it look like the field is missing, while the old one shows that it must be a positive number. The field is already required (cannot be missing), and if someone explicitly sets it to a non-positive number, the new error seems wrong to me.

Reply:
good catch, I need to wrap this error message.

Reply:
I don't like the old ones because they're all over the place, so trying to be more consistent.